src/plugin/NodeResolve.ts

```diff
@@ -441,7 +464,7 @@ export class NodeResolve implements LoaderPlugin {
 				if(!pkg) return Promise.reject(new Error('Error fetching ' + resolvedKey));
 
 				resolvedKey = loader.resolveSync(key, baseKey, ref);
-				return checkFile(loader, resolvedKey, key, baseKey, ref);
+				return checkFile(loader, resolvedKey, key, baseKey, ref!);
```

Comment:
There seem to be a lot of `var!`s, this is a bit of a code smell, perhaps we should update they types to actually represent the state?

Reply:
The compiler annoyingly fails to apply some type inference info from variables in outer functions to their references in inner functions. Fixed by adding default values to parameters at the cost of a slightly smaller minified result.